DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/IncidentConsumer.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System.IO;
+using System.Net;
+using System.Reflection;
+using System.Windows;
+using System;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace IncidentConsumer
+{
+    public class IncidentConsumer
+    {
+        string TenantId = Environment.GetEnvironmentVariable("TenantId");
+        string ClientId = Environment.GetEnvironmentVariable("ClientId");
+        string ClientKey = Environment.GetEnvironmentVariable("ClientKey");
+        const string ARM_ENDPOINT = "https://management.azure.com/";
+
+        /*
+         * need to rewrite GetAccessTokenAsync function
+         * as it uses obsolete technology to get the bearer token.
+         */
+        internal async Task<string> GetAccessTokenAsync(string uri)
+        {
+            var credential = new ClientCredential(ClientId, ClientKey);
+            var authenticationContext = new AuthenticationContext($"https://login.microsoftonline.com/{TenantId}");
+            var result = await authenticationContext.AcquireTokenAsync(uri, credential);
+            return result.AccessToken;
+        }
+
+        private string doPUT(string URI, string body, String token, ILogger log)
+        {
+            Uri uri = new Uri(String.Format(URI));
+
+            // Create the request
+            var httpWebRequest = (HttpWebRequest) WebRequest.Create(uri);
+            httpWebRequest.Headers.Add(HttpRequestHeader.Authorization, "Bearer " + token);
+            httpWebRequest.ContentType = "application/json";
+            httpWebRequest.Method = "PUT";
+
+            try
+            {
+                using (var streamWriter = new StreamWriter(httpWebRequest.GetRequestStream()))
+                {
+                    streamWriter.Write(body);
+                    streamWriter.Flush();
+                    streamWriter.Close();
+                }
+            }
+            catch (Exception ex)
```

Comment:
Then catch this exception at the end of the function and print it out there. The whole purpose of using try-catch instead of if-then-else.is to stop handling each and every case and make the code less branchy and more readable.
Inside the try-block, you put a happy path, and then you catch all errors and do all debug print outs in the catch block.

Reply:
how about now?